### PR TITLE
REPLAT-1677/Support video lead assets for native components

### PR DESCRIPTION
**Pull request by fourlastor**
This PR introduces support for videos on lead assets on native components.

I split the lead asset in 2 components: `ArticleLeadAssetImage` and `ArticleLeadAssetVideo` which will be responsible for rendering the correct version of a lead asset.

Major changes are:

- Web lead asset version is now `article-lead-asset.web.js`
- `getLeadAsset` includes `isVideo` in data as that's the only bit passed to `renderRow`

### CODE REVIEW

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/f47c658fd564ddbd16b08bb1711073ade5439610/index.html

File: packages/article/article-lead-asset-video.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    </Button>
+  );
+};
+
+export const propTypes = {
+  brightcoveVideoId: PropTypes.string.isRequired,
+  brightcovePolicyKey: PropTypes.string.isRequired,
+  brightcoveAccountId: PropTypes.string.isRequired,
+  posterImage: PropTypes.shape({
+    crop: PropTypes.shape({
```
Comment (colincclark) [inline packages/article/article-lead-asset-video.js]: can you DRY up these prop types by importing from a proptypes file?

File: packages/article/article-lead-asset.web.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      />
+    );
+  } else {
+    leadAssetView = <Image uri={displayImage.url} />;
+  }
+
+  return (
+    <LeadAsset>
+      <AspectRatioContainer aspectRatio={aspectRatio}>
+        {leadAssetView}
```
Comment (colincclark) [inline packages/article/article-lead-asset.web.js]: can this be a function to negate the need for reassignment of `leadAssetView`?

Comment (fourlastor) [reply packages/article/article-lead-asset.web.js]: dunno, I literally renamed the file, not much than that

File: packages/article/article-lead-asset-video.js (hunk 1 above)
Comment (fourlastor) [reply packages/article/article-lead-asset-video.js]: something like `crop-prop-types.js`?

Comment (fourlastor) [reply packages/article/article-lead-asset-video.js]: 4d12dac

File: packages/article/article-lead-asset.web.js (hunk 2 above)
Comment (fourlastor) [reply packages/article/article-lead-asset.web.js]: Ah ok, I see what you meant: cb18ff1

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/cb18ff1942cce5f0ca47e3569157ef69f383fde2/index.html

**Pull request merged**